### ISSUE TITLE
Todo: Refactor `ObjectInputStream` to use std::stringstream

### ISSUE THREAD

**Issue by Febbe** [xournalpp/xournalpp#2278]
The `ObjectInputStream` class uses a self implemented serialization mechanism.
 - The read data is `reinterpret_cast`ed to the read values this results in a wrong alignment this is UB. 
 - Currently prone to bugs and should be replaced with a correct implementation like std::stringstream.

```text
ObjectInputStream.cpp

Load of misaligned address 0x6190005ad782 for type 'int', which requires 4 byte alignment
    ObjectInputStream::readString() ObjectInputStream.cpp:110
Load of misaligned address 0x6190005ad7bf for type 'double', which requires 8 byte alignment
    ObjectInputStream::readDouble() ObjectInputStream.cpp:86
Load of misaligned address 0x6190005ad88f for type 'int', which requires 4 byte alignment
    ObjectInputStream::readInt() ObjectInputStream.cpp:74
Load of misaligned address 0x6190005ad89d for type 'size_t' (aka 'unsigned long'), which requires 8 byte alignment
    ObjectInputStream::readSizeT() ObjectInputStream.cpp:98
Load of misaligned address 0x6190005ad8bf for type 'int', which requires 4 byte alignment
    ObjectInputStream::readData(void**, int*) ObjectInputStream.cpp:129
Load of misaligned address 0x6190005ad8c3 for type 'int', which requires 4 byte alignment
    ObjectInputStream::readData(void**, int*) ObjectInputStream.cpp:132
    
    Stroke::readSerialized(ObjectInputStream&) : Stroke.cpp:68
    Control::clipboardPasteXournal(ObjectInputStream&) Control.cpp:2638
```

**Comment by Febbe** [comment]
May cause #2262

**Comment by theanshm** [comment]
I would like to work on this issue if it is open to work on. I am new to contributing to open source and thus would require a little push in the right direction but I am willing to learn.

**Comment by bhennion** [comment]
One remark: from the current version of the code in `Stroke::readSerialized` (and probably elsewhere too)
```
    in.readData(reinterpret_cast<void**>(&p), &count);
    this->points = std::vector<Point>{p, p + count};
```
The `ObjectInputStream::readData` method is not compatible with modern containers. This results in the data being copied twice on deserialization. This probably doubles the cost of the process for no reason. This should be fixed along with the above.

**Comment by Febbe** [comment]
Closed by #2870
